qemu/qmp.go

```diff
@@ -602,16 +602,26 @@ func (q *QMP) ExecuteQuit(ctx context.Context) error {
 // used to name the device.  As this identifier will be passed directly to QMP,
 // it must obey QMP's naming rules, e,g., it must start with a letter.
 func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string) error {
-	args := map[string]interface{}{
-		"options": map[string]interface{}{
-			"driver": "raw",
-			"file": map[string]interface{}{
-				"driver":   "file",
-				"filename": device,
-			},
-			"id": blockdevID,
+	var args map[string]interface{}
+
+	blockdevArgs := map[string]interface{}{
+		"driver": "raw",
+		"file": map[string]interface{}{
+			"driver":   "file",
+			"filename": device,
 		},
 	}
+
+	if q.version.Major == 2 && q.version.Minor >= 9 {
```

Comment:
This should probably be 

if q.version.Major > 2 || (q.version.Major == 2 && q.version.Minor >= 9)

Otherwise, we'll fall back to the old protocol when qemu 3.0 comes out which is probably not what we want.